### PR TITLE
Animation doc markup cleanups.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/animation.py [hunk 1]
```diff
@@ -253,19 +255,19 @@ def __init__(self, fps=5, codec=None, bitrate=None, extra_args=None,
...
+            implies that the bitrate should be determined automatically by the
+            underlying utility.
+        extra_args: list of strings or None, optional
             A list of extra string arguments to be passed to the underlying
-            movie utility. The default is None, which passes the additional
-            arguments in the 'animation.extra_args' rcParam.
-        metadata: dict of string:string or None
+            movie utility. The default is ``None``, which passes the additional
+            arguments in the ``animation.extra_args`` rcParam.
+        metadata: Dict[str, str] or None
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: optional?

File: lib/matplotlib/animation.py [hunk 2]
```diff
@@ -964,56 +965,53 @@ def save(self, filename, writer=None, fps=None, dpi=None, codec=None,
...
-            class to use, such as 'ffmpeg' or 'mencoder'. If `None`,
-            defaults to ``rcParams['animation.writer']``
+            class to use, such as 'ffmpeg' or 'mencoder'. If ``None``,
+            defaults to ``rcParams['animation.writer']``.
 
         fps : number, optional
-           frames per second in the movie. Defaults to None,
-           which will use the animation's specified interval to set
-           the frames per second.
+           frames per second in the movie. Defaults to None, which will use the
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: \`\`None\`\`

File: lib/matplotlib/animation.py [hunk 3]
```diff
@@ -1313,15 +1311,15 @@ class TimedAnimation(Animation):
 
     repeat_delay : number, optional
         If the animation in repeated, adds a delay in milliseconds
-        before repeating the animation.  Defaults to `None`.
+        before repeating the animation.  Defaults to ``None``.
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Remove extra space after period?

File: lib/matplotlib/animation.py [hunk 4]
```diff
@@ -1313,15 +1311,15 @@ class TimedAnimation(Animation):
...
 
     repeat : bool, optional
         Controls whether the animation should repeat when the sequence
-        of frames is completed. Defaults to `True`.
+        of frames is completed.  Defaults to ``True``.
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Why add the extra space?

File: lib/matplotlib/animation.py [hunk 5]
```diff
@@ -1402,15 +1399,15 @@ class ArtistAnimation(TimedAnimation):
 
     repeat_delay : number, optional
         If the animation in repeated, adds a delay in milliseconds
-        before repeating the animation.  Defaults to `None`.
+        before repeating the animation.  Defaults to ``None``.
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Remove extra space after period?

File: lib/matplotlib/animation.py [hunk 6]
```diff
@@ -1475,61 +1471,60 @@ class FuncAnimation(TimedAnimation):
...
-       The number of values from `frames` to cache.
+       The number of values from *frames* to cache.
 
     interval : number, optional
        Delay between frames in milliseconds.  Defaults to 200.
 
     repeat_delay : number, optional
        If the animation in repeated, adds a delay in milliseconds
-       before repeating the animation.  Defaults to `None`.
+       before repeating the animation.  Defaults to ``None``.
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Extra space.

File: lib/matplotlib/animation.py [hunk 7]
```diff
@@ -1475,61 +1471,60 @@ class FuncAnimation(TimedAnimation):
...
 
     repeat : bool, optional
        Controls whether the animation should repeat when the sequence
-       of frames is completed.  Defaults to `True`.
+       of frames is completed.  Defaults to ``True``.
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Extra space.

File: lib/matplotlib/animation.py (hunk 1 above)
Comment (anntzer) [inline lib/matplotlib/animation.py]: I think `dict of string:string` benefits from being changed into `Dict[str,str]` but `... or None` is so prevalent throughout the code base that such a change (if done) should be the object of another PR.

File: lib/matplotlib/animation.py (hunk 3 above)
Comment (anntzer) [inline lib/matplotlib/animation.py]: I tend to prefer two spaces after a period (https://en.wikipedia.org/wiki/Sentence_spacing -- I know there are arguments both ways).  I don't think it's worth changing the style throughout the docs, but at least locally it should be better to uniformize it (so here I settled towards two-spacing).  Same reply applies to comments below.

File: lib/matplotlib/animation.py (hunk 2 above)
Comment (anntzer) [inline lib/matplotlib/animation.py]: fixed

File: lib/matplotlib/animation.py (hunk 3 above)
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Fair enough.

File: lib/matplotlib/animation.py (hunk 1 above)
Comment (dopplershift) [inline lib/matplotlib/animation.py]: You changed the one above, so I wasn't sure if you wanted to be consistent.

**Comment by dopplershift** [comment]
Waiting to see if we can get Travis green here...
